linkhash.c

```diff
@@ -577,9 +580,12 @@ int lh_table_insert_w_hash(struct lh_table *t, const void *k, const void *v, con
 {
 	unsigned long n;
 
-	if (t->count >= t->size * LH_LOAD_FACTOR)
-		if (lh_table_resize(t, t->size * 2) != 0)
+	if (t->count >= t->size * LH_LOAD_FACTOR) {
+		/* Avoid signed integer overflow with large tables. */
+		int new_size = INT_MAX / 2 < t->size ? t->size * 2 : INT_MAX;
+		if (lh_table_resize(t, new_size) != 0)
```

Comment:
This needs to either avoid calling lh_table_resize() when we're already at INT_MAX size, or simply fail once we've exceeded the load factor at that point.  As it is, we're going to be needlessly reallocating and recalculating all (INT_MAX * LH_LOAD_FACTOR) entries on every insert.

Reply:
Oh I see what you mean now. Thanks for pointing out. I have adjusted the code to check if size is already INT_MAX.